apps/election-manager/src/components/ExportFinalResultsModal.tsx

```diff
@@ -0,0 +1,299 @@
+            </p>
+          </Prose>
+        }
+        onOverlayClick={onClose}
+        actions={actions}
+      />
+    )
+  }
+
+  if (currentState === ModalState.SAVING) {
+    return (
+      <Modal isOpen={isOpen} content={<Loading />} onOverlayClick={onClose} />
+    )
+  }
+
+  if (currentState !== ModalState.INIT) {
+    throwBadStatus(currentState) // Creates a compile time check that all states are being handled.
+  }
+
+  switch (usbDriveStatus) {
+    case UsbDriveStatus.absent:
+    case UsbDriveStatus.notavailable:
+    case UsbDriveStatus.recentlyEjected:
+      // When run not through kiosk mode let the user download the file
+      // on the machine for internal debugging use
+      return (
+        <Modal
+          isOpen={isOpen}
+          content={
+            <Prose>
+              <h1>No USB Drive Detected</h1>
+              <p>
+                <USBImage src="usb-drive.svg" alt="Insert USB Image" />
+                Please insert a USB drive in order to export the scanner
+                results.
+              </p>
+            </Prose>
+          }
+          onOverlayClick={onClose}
+          actions={
+            <React.Fragment>
+              <LinkButton onPress={onClose}>Cancel</LinkButton>
+              {!window.kiosk && (
+                <Button
+                  data-testid="manual-export"
+                  onPress={() => exportResults()}
+                >
+                  Export
+                </Button>
+              )}{' '}
+            </React.Fragment>
+          }
+        />
+      )
+    case UsbDriveStatus.ejecting:
+    case UsbDriveStatus.present:
+      return (
+        <Modal
+          isOpen={isOpen}
+          content={<Loading />}
+          onOverlayClick={onClose}
+          actions={
+            <React.Fragment>
+              <LinkButton onPress={onClose}>Cancel</LinkButton>
+            </React.Fragment>
+          }
+        />
+      )
+    case UsbDriveStatus.mounted:
+      return (
+        <Modal
+          isOpen={isOpen}
+          content={
+            <MainChild>
+              <Prose>
+                <h1>Export Results</h1>
+                <p>
+                  Export the final tally results to the following filename on
+                  the inserted USB drive.
+                </p>
+              </Prose>
+              <LabelText small>Filename</LabelText>
+              <TextInputRow>
+                <FilenameInput
+                  data-testid="filename-input"
+                  value={filename}
+                  onChange={handleFilenameInputChange}
+                />
+                <span>.csv</span>
+              </TextInputRow>
+            </MainChild>
+          }
+          onOverlayClick={onClose}
+          actions={
+            <React.Fragment>
+              <LinkButton onPress={onClose}>Cancel</LinkButton>
+              <Button primary onPress={() => exportResults()}>
```

Comment:
Any reason this is an inline function and the `onClose` above is not? Seems like this could be just `onPress={exportResults}`.